src/url/cloudinaryURL.ts

```diff
@@ -7,9 +9,60 @@ import Transformation from '../transformation/Transformation';
  * @param {Object} descriptor
  * @param {Transformation} transformation
  */
-function createCloudinaryURL(config: unknown, descriptor: unknown, transformation: unknown) {
-  // Currently not implemented
-  return '';
+function createCloudinaryURL(config: Config, descriptor?: Descriptor, transformation?: Transformation) {
+  const prefix = getUrlPrefix(config.cloudName);
+  const resourceType = handleResourceType(descriptor);
+  const type = handleType(descriptor);
+  const transformationString = transformation ? transformation.toString() : '';
+  const publicId = config.publicId;
+
+  const url = [prefix, resourceType, type, transformationString, publicId]
+    .join('/')
+    .replace(/([^:])\/+/g, '$1/') // replace '///' with '//'
+    .replace(' ', '%20');
+
+  return url;
+}
+
+/**
+ * Create the URL prefix for Cloudinary resources.
+ * @param cloudName
+ */
+function getUrlPrefix(cloudName: string) {
```

Comment:
I never liked this function in the original code as well,
the prefix is actually quite a few things combined

https://res.cloudinary.com/demo

I feel calling it 'prefix' is too wide, maybe "getFullCloudPath" or something along these lines?

it's not clear if "URL prefix" is https:// or https://res.cloudianry or anything else.